packages/itmat-ui-react/src/components/users/userDetails.tsx

```diff
@@ -213,29 +213,34 @@ export const EditUserForm: React.FunctionComponent<{ user: (IUserWithoutToken &
                                 : null
                             }
                             &nbsp;&nbsp;&nbsp;
-                            <Mutation<any, any>
-                                mutation={DELETE_USER}
-                                refetchQueries={[
-                                    { query: GET_USERS, variables: { fetchDetailsAdminOnly: false, fetchAccessPrivileges: false } },
-                                    /* quick fix: TO_DO, change to cache modification later */
-                                    ...(user.access!.studies.map(el => ({ query: GET_STUDY, variables: { studyId: el.id } })))
-                                ]}
-                            >
+                            {whoamidata.whoAmI.id !== user.id && whoamidata.whoAmI.type === userTypes.ADMIN
+                                ? <>
+                                    <Mutation<any, any>
+                                        mutation={DELETE_USER}
+                                        refetchQueries={[
+                                            { query: GET_USERS, variables: { fetchDetailsAdminOnly: false, fetchAccessPrivileges: false } },
+                                            /* quick fix: TO_DO, change to cache modification later */
+                                            ...(user.access!.studies.map(el => ({ query: GET_STUDY, variables: { studyId: el.id } })))
+                                        ]}
+                                    >
```

Comment:
This check wouldn't be done in the UI, it is a server-side check.
I would also ask that this be placed in a separate PR because it tackles a very different issue 😅